snuba/consumers/consumer_builder.py

```diff
@@ -93,12 +95,15 @@ def __init__(
             .topic
         )
 
+        validate_passed_slice(storage_key, slice_id)
```

Comment:
I think we should also move this call to the centralized code so it happens automatically on an attempt to load the storage and does not require the user to explicitly call it.

Reply:
Where are you referring to by "centralized code"?

Reply:
The code which you linked is currently only used for the QueryPlanBuilder, and doesn't actually directly load the storages. The Consumer Builder only uses `get_writable_storage` to get and load the storage from config, but we decided not to make that path slice-aware.

Reply:
Personally I think that there may not be a very obvious place. I think the reason for this is might be twofold:
1. storage objects are unsliced
2. slicing is not really even done at the storage level, it's done at the storage set level

So I think this is fine if you want to leave this here now. But I'm not sure if this setting that we are validating against is actually the right one given that slices are generally defined against storage sets and not the storages themselves.

Reply:
I think what I'll do is leave this here for now, but make a note to revisit the location for this when we do the rename from `SLICED_STORAGES` to `SLICED_STORAGE_SETS`?